libsolidity/inlineasm/AsmCodeGen.cpp

```diff
@@ -197,7 +185,30 @@ class CodeTransform: public boost::static_visitor<>
 	void operator()(Label const& _label)
 	{
 		m_state.assembly.setSourceLocation(_label.location);
-		m_state.assembly.append(m_state.labels.at(_label.name));
+		solAssert(m_scope.identifiers.count(_label.name), "");
+		Scope::Label& label = boost::get<Scope::Label>(m_scope.identifiers[_label.name]);
+		assignLabelIdIfUnset(label);
+		if (label.resetStackHeight)
+		{
+			size_t numVariables = boost::range::count_if(
+				 m_scope.identifiers | boost::adaptors::map_values,
+				[](Scope::Identifier const& var) { return var.type() == typeid(Scope::Variable) && boost::get<Scope::Variable>(var).active; }
+			);
+			numVariables += boost::count_if(_label.stackInfo, [](string const& s) { return !s.empty(); });
+			m_state.assembly.setDeposit(m_initialDeposit + numVariables);
+		}
+		else if (label.stackAdjustment != 0)
+			m_state.assembly.adjustDeposit(label.stackAdjustment);
+		int height = m_state.assembly.deposit();
+		for (auto const& identifier: _label.stackInfo | boost::adaptors::reversed)
+			if (!identifier.empty())
+			{
+				solAssert(m_scope.identifiers.count(identifier), "");
+				Scope::Variable& var = boost::get<Scope::Variable>(m_scope.identifiers[identifier]);
+				var.active = true;
+				var.stackHeight = --height;
```

Comment:
Why does the height decrease instead of increase?  I guess a new identifier is being added here.